version: '3.8'

services:
  web:
    build:
      context: .
      dockerfile: ./compose/local/flask/Dockerfile
    image: callmebot
    command: /start
    volumes: 
      - .:/application
    ports:
      - "5010:5000"
    env_file:
      - .env
    depends_on:
      - redis
      - db
    networks:
      - callmebot-network

  db: 
    image: postgres:13-alpine
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_DB=callmebot
      - POSTGRES_USER=callmebot
      - POSTGRES_PASSWORD=callmebot
    ports:
      - "5432:5432"
    networks:
      - callmebot-network

  redis:
    image: redis:6-alpine
    networks:
      - callmebot-network

  celery_worker:
      build:
        context: .
        dockerfile: ./compose/local/flask/Dockerfile
      image: callmebot_celery_worker
      command: /start-celeryworker
      volumes:
        - .:/application
      env_file:
        - .env
      depends_on:
        - redis
        - db
      networks:
      - callmebot-network
      
  celery_beat:
    build:
      context: .
      dockerfile: ./compose/local/flask/Dockerfile
    image: callmebot_celery_beat
    command: /start-celerybeat
    volumes:
      - .:/application
    env_file:
      - .env
    depends_on:
      - redis
      - db
    networks:
      - callmebot-network

volumes:
  postgres_data:

networks:
  callmebot-network:
    driver: bridge